mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -24,11 +25,55 @@ def SYCL_Dialect : Dialect {
   let name = "sycl";
   let cppNamespace = "::mlir::sycl";
   let useDefaultTypePrinterParser = 1;
+  let extraClassDeclaration = [{
+    llvm::DenseMap<std::pair<::mlir::TypeID, llvm::StringRef>, llvm::StringRef>
+        Methods;
+
+    static llvm::StringRef getBaseTypeAttrName() { return "BaseType"; }
+
+    static llvm::StringRef getBaseTypeNameAttrName() { return "BaseTypeName"; }
+
+    static llvm::StringRef getFunctionNameAttrName() { return "Function"; }
+
+    static llvm::StringRef getMangledFunctionNameAttrName() { return "MangledName"; }
+
+    void addSYCLMethod(
+        mlir::TypeID TypeID, llvm::ArrayRef<llvm::StringLiteral> MethodNames,
+        llvm::StringRef OpName);
+
+    template <typename T> void addSYCLMethod();
+
+    template <typename... Args> void addOperations();
+
+    llvm::Optional<llvm::StringRef> findMethod(::mlir::TypeID Type,
```

Comment:
I would describe that is `Type` (a SYCL class type ?). What is name (the unmangled name of a SYCL method ? Maybe call it methodName). Will this function return the mangled name of the SYCL class method it found (when it finds one)?

Reply:
It shall return the name of the SYCL dialect operation implementing the method. I'll make the changes and add a comment explaining it